src/services/AbstractDFG.jl

```diff
@@ -84,82 +87,92 @@ end
 """
     $(SIGNATURES)
 Get a DFGVariable from a DFG using its underlying integer ID.
+Returns: DFGVariable if available, nothing if not.
 """
-function getVariable(dfg::G, variableId::Int64)::AbstractDFGVariable where G <: AbstractDFG
+function getVariable(dfg::G, variableId::Int64)::Union{Nothing, AbstractDFGVariable} where G <: AbstractDFG
 	error("getVariable not implemented for $(typeof(dfg))")
 end
 
 """
     $(SIGNATURES)
 Get a DFGVariable from a DFG using its label.
+Returns: DFGVariable if available, nothing if not.
 """
-function getVariable(dfg::G, label::Union{Symbol, String})::AbstractDFGVariable where G <: AbstractDFG
+function getVariable(dfg::G, label::Union{Symbol, String})::Union{Nothing, AbstractDFGVariable} where G <: AbstractDFG
 	return getVariable(dfg, Symbol(label))
 end
 
 """
     $(SIGNATURES)
 Get a DFGFactor from a DFG using its underlying integer ID.
+Returns: DFGFactor if available, nothing if not.
 """
-function getFactor(dfg::G, factorId::Int64)::AbstractDFGFactor where G <: AbstractDFG
+function getFactor(dfg::G, factorId::Int64)::Union{Nothing, AbstractDFGFactor} where G <: AbstractDFG
 	error("getFactor not implemented for $(typeof(dfg))")
 end
 
 """
     $(SIGNATURES)
 Get a DFGFactor from a DFG using its label.
+Returns: DFGFactor if available, nothing if not.
 """
-function getFactor(dfg::G, label::Union{Symbol, String})::AbstractDFGFactor where G <: AbstractDFG
+function getFactor(dfg::G, label::Union{Symbol, String})::Union{Nothing, AbstractDFGFactor} where G <: AbstractDFG
 	return getFactor(dfg, Symbol(label))
 end
 
 """
     $(SIGNATURES)
 Update a complete DFGVariable in the DFG.
+Returns: True if updated, false if not.
 """
-function updateVariable!(dfg::G, variable::V)::AbstractDFGVariable where {G <: AbstractDFG, V <: AbstractDFGVariable}
+function updateVariable!(dfg::G, variable::V)::Bool where {G <: AbstractDFG, V <: AbstractDFGVariable}
 	error("updateVariable! not implemented for $(typeof(dfg))")
 end
 
 """
     $(SIGNATURES)
 Update a complete DFGFactor in the DFG.
+Returns: True if updated, false if not.
 """
-function updateFactor!(dfg::G, factor::F)::AbstractDFGFactor where {G <: AbstractDFG, F <: AbstractDFGFactor}
+function updateFactor!(dfg::G, factor::F)::Bool where {G <: AbstractDFG, F <: AbstractDFGFactor}
 	error("updateFactor! not implemented for $(typeof(dfg))")
 end
 
 """
     $(SIGNATURES)
 Delete a DFGVariable from the DFG using its label.
+Returns: True if deleted, false if not.
 """
-function deleteVariable!(dfg::G, label::Symbol)::AbstractDFGVariable where G <: AbstractDFG
+function deleteVariable!(dfg::G, label::Symbol)::Bool where G <: AbstractDFG
```

Comment:
did we come to a decision delete being bool or union in the end?

Reply:
maybe there is a pointer to the decision on this?  but my expectation would be give the just deleted value back one last time...